src/Components/BehaviorsModal/Internal/Widgets/DataHistoryWidget/DataHistoryWidget.tsx

```diff
@@ -1,47 +1,178 @@
+            <HighChartsWrapper
+                title={displayName}
+                seriesData={highChartSeriesData}
+                isLoading={isLoading}
+                chartOptions={{
+                    titleTargetLink:
+                        mode === BehaviorModalMode.viewer
+                            ? deeplink
+                            : undefined,
+                    legendLayout: 'vertical',
+                    legendPadding: 0,
+                    hasMultipleAxes: chartOptions.yAxisType === 'independent',
+                    dataGrouping: chartOptions.aggregationType
+                }}
+            />
         </div>
     );
 };
+
+/** Gets timeSeries list from data history widget and twins and
+ * evaluate the twin id and property name tuples using the resolved twins to use in query
+ */
+const getTwinIdPropertyMap = (
+    timeSeries: IDataHistoryTimeSeries,
+    twins: Record<string, DTwin>
+): Array<IDataHistoryWidgetTimeSeriesTwin> =>
+    twins
+        ? timeSeries.map((ts) => {
+              const aliasAndPropertyTuple = ts.expression?.split('.'); // TODO: change this logic when timeSeries expressions gets complex, current expression is in [PrimaryTwin.Temperature] format
+              if (twins && aliasAndPropertyTuple.length === 2) {
+                  return {
+                      label: ts.label,
+                      twinId: twins[aliasAndPropertyTuple[0]]?.$dtId,
+                      twinPropertyName: aliasAndPropertyTuple[1]
+                  };
+              }
+          })
+        : null;
+
+/** Gets fetched adx time series data and data history widget time series to twin mapping information
+ * to get the labels if defined for each series, and converts it into high chart series data to render in chart
+ */
+const transformADXTimeSeriesToHighChartsSeries = (
+    adxTimeSeries: Array<ADXTimeSeries>,
+    twinIdPropertyMap: Array<IDataHistoryWidgetTimeSeriesTwin>
+): Array<IHighChartSeriesData> =>
+    adxTimeSeries && twinIdPropertyMap
+        ? adxTimeSeries.map(
+              (series) =>
+                  ({
+                      name:
+                          twinIdPropertyMap.find(
+                              (map) =>
+                                  map.twinId === series.id &&
+                                  map.twinPropertyName === series.key
+                          )?.label || series.id + ' ' + series.key, // this is the label for series to show in chart
+                      data: series.data
+                  } as IHighChartSeriesData)
+          )
+        : [];
+
+/** Generate placeholder mock data for timeseries to show in chart in preview mode
+ */
+const generatePlaceholderHighChartsData = (
+    widgetConfig: IDataHistoryWidgetConfiguration,
+    placeholderTimeSeriesData: Array<Array<TimeSeriesData>>
+): Array<IHighChartSeriesData> => {
+    return widgetConfig.timeSeries.map((ts, idx) => ({
+        name: `${ts.label || ts.expression}`,
```

Comment:
we cannot evaluate the expression in Builder mode since we dont have resolved twin information (which we have from useRuntimeSceneData hook in the viewer mode)

Reply:
but this is on teh viewer side not builder right?

Reply:
this is part of generatePlaceholderHighChartsData method which is called when it is in preview mode which is builder